curl/tests/client_test.cc

```diff
@@ -469,15 +473,15 @@ namespace leatherman { namespace curl {
              test_impl->test_failure_mode = curl_impl::error_mode::set_url_error;
              REQUIRE_THROWS_AS(test_client.download_file(req, file_path), curl::http_file_download_exception);
```

Comment:
Server connection failures would result in two exceptions being thrown:

- http_request
- http_download

so if you want to isolate them, you can just catch either http_request or http_download, ignoring http_setup completely. If you care about whether something went wrong in the setup, then you'd need to catch setup before the request/download.

The main distinction between request and download is that download has a file. If the download_file call were nested deep in a bunch of method calls, and only the top-level caller has the try-catch block, then it is possible that you can't get the path of the file as the caller when download_file throws an exception. That's why I think it would help for the exception to contain that information.